src/lib/server/endpoints/images.ts

```diff
@@ -0,0 +1,214 @@
+import type { Sharp } from "sharp";
+import sharp from "sharp";
+import type { MessageFile } from "$lib/types/Message";
+import { z } from "zod";
+
+export function createImageProcessorOptionsValidator<MimeType extends string = string>(
```

Comment:
```suggestion
export function createImageProcessorOptionsValidator<TMimeType extends string = string>(
```

rename `MimeType` to `TMimeType` to signify that it is a generic